driver-core/src/main/com/mongodb/internal/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -168,15 +168,22 @@ public void onResult(final List<RawBsonDocument> rawDocuments, final Throwable t
                     List<T> results = new ArrayList<T>();
                     for (RawBsonDocument rawDocument : rawDocuments) {
                         if (!rawDocument.containsKey("_id")) {
-                            callback.onResult(null,
-                                    new MongoChangeStreamException("Cannot provide resume functionality when the resume token is missing.")
+                            callback.onResult(
+                                null,
+                                new MongoChangeStreamException(
+                                    "Cannot provide resume functionality when the resume token is missing."
+                                )
                             );
                             return;
                         }
                         results.add(rawDocument.decode(changeStreamOperation.getDecoder()));
```

Comment:
I think this decode method needs to be in the try / catch.

Reply:
ohh.. sorry. In the last commit I have hurry up a bit.. 
All fixed now =)